azure/migrate-resource.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+set -e
+############################################################
+# Remove and import a Terraform resource from the tfstate
+############################################################
+# Global variables
+VERS="1.0"
+
+############################################################
+# Print usage information                                                     #
+############################################################
+function print_usage() {
+  me=`basename "$0"`
+  echo "Setup v."${VERS} "sets up a configuration relative to a specific subscription"
+  echo "  ./${me} <resource name to remove> <resource name to add> <ENV>"
+  for thisenv in $(ls "env/")
+  do
+      echo "  Example: ./${me} \"module.function_lollipop[0].azurerm_function_app.this\" \"module.function_lollipop[0].azurerm_linux_function_app.this\" ${thisenv}"
+  done
+  echo
+  echo "Syntax: setup.sh [-l|h]"
+  echo "  options:"
+  echo "  h     Print this Help."
+  echo "  l     List available environments."
+  echo
+}
+
+function removeAndImport() {
```

Comment:
Do we need this function? Is it doing this refactoring provided by terraform itself?
https://developer.hashicorp.com/terraform/language/modules/develop/refactoring

Reply:
They are in fact pretty similar but consider "moved blocks" functionality has some limitations; for example: Move statements can be only within a single module package (https://nedinthecloud.com/2021/12/14/using-the-moved-block-in-terraform-1-1/)
Furthermore, consider it is not perfect (e.g. https://github.com/hashicorp/terraform/issues/32758 "Moved block failure after import"), so the reason for using this script is that you can customize the rename (aka removeandimport) via bash script or cli commands.